lib/internal/Magento/Framework/HTTP/PhpEnvironment/RemoteAddress.php

```diff
@@ -99,9 +104,9 @@ function (string $ip) {
                     return !in_array(trim($ip), $this->trustedProxies, true);
                 }
             );
-            $remoteAddress = trim(array_pop($ipList));
+            $remoteAddress = trim((string) array_pop($ipList));
         } else {
-            $remoteAddress = trim(reset($ipList));
+            $remoteAddress = trim((string) reset($ipList));
```

Comment:
Can we revert this? The goal on this one is literally just to mark classes/interfaces as API and not make any code change. If this is required because of `strict_types=1`, lets remove it and create a separate PR for it.